src/FourBitTwoDisclosureDeviceUnlocker.java

```diff
@@ -73,11 +85,26 @@ public static boolean unlock(final Device dev) {
             }
             n--;
         }
-        return false;
+        if (!isUnlocked) {
+            n = numOfBits;
+            while (n > 0) {
+                for (CharSequence perm : perms) {
+                    doPeek(perm);
+                    doPoke();
+                    // we want a random number of spins from 1 to the number of bits. 0 spins is invalid doSpin()
+                    isUnlocked = doSpin((int)Math.round(Math.random() * numOfBits) + 1);
+                    if (isUnlocked) {
+                        break;
+                    }
+                }
+                n--;
+            }
+        }
+        return isUnlocked;
     }
```

Comment:
This is the added heuristic while loop to hopefully unlock devices that happen to stay locked after our first while loop, by randomly spinning before each `peek` and `poke`.